base/base/getMemoryAmount.cpp

```diff
@@ -11,6 +13,54 @@
 #endif
 
 
+namespace
+{
+
+std::optional<uint64_t> getCgroupsV2MemoryLimit([[maybe_unused]] const std::string & setting)
+{
+#if defined(OS_LINUX)
+    /// Identify the cgroup the process belongs to
+    std::ifstream cgroup_name_file("/proc/self/cgroup");
+    if (!cgroup_name_file.is_open())
+        return {};
+
+    std::string cgroup_name;
+    cgroup_name_file >> cgroup_name;
+
+    /// cgroups v2 will show a single line with prefix "0::/"
+    /// - https://book.hacktricks.xyz/linux-hardening/privilege-escalation/docker-security/cgroups
+    const std::string v2_prefix = "0::/";
+    if (!cgroup_name.starts_with(v2_prefix))
+        return {};
+    cgroup_name = cgroup_name.substr(v2_prefix.length());
```

Comment:
is it guaranteed that there is only a single name?

Reply:
Yes, there is such a guarantee for v2.

It is (un?)fortunately possible to have Linux hosts with mixed v1/v2 usage (see the link to book.hacktricks.xyz), in this case there would be more lines. But afaik, that was only during a short transition period in pre-2020 the case.

Reply:
But perhaps I should add a check to make sure the system is v2-only.